static/app/components/charts/series/barSeries.tsx

```diff
@@ -1,11 +1,15 @@
 import 'echarts/lib/chart/bar';
 
-import type {BarSeriesOption} from 'echarts';
+import type {BarSeriesOption, LineSeriesOption} from 'echarts';
 
-function barSeries(props: BarSeriesOption): BarSeriesOption {
+/**
+ * The return type can be BarSeriesOption or LineSeriesOption so that we can add
+ * custom lines on top of the event bar chart in `eventGraph.tsx`.
+ */
+function barSeries(props: BarSeriesOption): BarSeriesOption | LineSeriesOption {
   return {
     ...props,
-    type: 'bar',
+    type: props.type ?? 'bar',
```

Comment:
is this change used somewhere?

also, `barSeries({type: line})` seems weird, how about `lineSeries()` for that case?

Reply:
yup, i pass in `type: line` in `flagSeries.tsx` to ensure the series gets counted as a non-bar-chart. otherwise, the event bars will shrink/grow when the flag series is hidden.

i have to keep `barSeries` to keep the types happy (`eventGraph.tsx` uses `BarChart` which makes sense for the default event bar chart rendering, and `BarChart` only accepts `BarSeries`.) i'm basically hacking the bar chart a little to let it accept stuff that's either type `bar` or type `line`.